x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -492,6 +492,39 @@ export class KnowledgeBaseService {
     };
   };
 
+  getInstructions = async (): Promise<UserInstruction[]> => {
+    try {
+      const response = await this.dependencies.esClient.search<KnowledgeBaseEntry>({
+        index: this.dependencies.resources.aliases.kb,
+        query: {
+          bool: {
+            must: [
+              {
+                term: {
+                  'labels.category.keyword': {
+                    value: 'instruction',
+                  },
+                },
+              },
+            ],
+          },
+        },
+        size: 500,
```

Comment:
should we cut off the size to a lower amount? maybe we cut off by token count and log a warning if the token count has been exceeded?

Reply:
I think cutting off by token count makes the most sense (allowing the user to freely split it into as many documents they wish, 500 hits is probably a safe limit).

Do you have any suggestion for how many tokens we want to allow for instructions?

Reply:
Something like 2000? Probably good to get the token count for the current system message (ignoring the APM part, we'll remove that) to get a sense of how long that is.

Reply:
Fixed in https://github.com/elastic/kibana/pull/180263/commits/09238d1932fb92cac98d462649aaa71b38195bb0